code/modules/mob/living/simple_animal/bot/secbot.dm

```diff
@@ -432,21 +432,37 @@
 
 
 /mob/living/simple_animal/bot/secbot/explode()
-	walk_to(src,0)
 	visible_message("<span class='userdanger'>[src] blows apart!</span>")
-	var/turf/Tsec = get_turf(src)
-	var/obj/item/secbot_assembly/Sa = new /obj/item/secbot_assembly(Tsec)
+	var/turf/explode_turf = get_turf(src)
+	var/obj/item/secbot_assembly/Sa = new /obj/item/secbot_assembly(explode_turf)
 	Sa.build_step = 1
 	Sa.overlays += "hs_hole"
 	Sa.created_name = name
-	new /obj/item/assembly/prox_sensor(Tsec)
-	new /obj/item/melee/baton(Tsec)
+	new /obj/item/assembly/prox_sensor(explode_turf)
+	new /obj/item/melee/baton(explode_turf)
 	if(prob(50))
-		drop_part(robot_arm, Tsec)
+		drop_part(robot_arm, explode_turf)
 	do_sparks(3, 1, src)
 	new /obj/effect/decal/cleanable/blood/oil(loc)
 	..()
 
+///Disassembling the bot in a civilized manner with a multitool
+/mob/living/simple_animal/bot/secbot/disassemble()
+	var/turf/disassemble_turf = get_turf(src)
+	var/obj/item/secbot_assembly/Sa = new /obj/item/secbot_assembly(disassemble_turf)
```

Comment:
Use a better variable name here, please. Preferably one that's camel_case.

Reply:
Guh- I hadn't noticed secbots have a `Sa` variable as well... I'll switch it to `assembly` like I did for the ed209

Reply:
Renamed it to `pierced_helmet` instead to avoid confusion while reading the code